### PR TITLE
Analytics key events client side tracking

### PR DESCRIPTION
**Pull request by JoanMora**
Adds visibility for upload and download file flows. 
Covers the start, completed, aborted and error states of downloads and uploads for files. 

Provides interfaces using the tracking plan.

Note that the following env variables need to be added, and the `.env.example` is updated accordingly.
```
# ANALYTICS KEYS
REACT_APP_ANALYTICS_ERROR_REPORTING_WRITE_KEY=
REACT_APP_CDP_DATA_PLANE=
````

### CODE REVIEW

File: src/app/drive/services/download.service/downloadFile.ts [hunk 1]
```diff
@@ -135,15 +124,19 @@ export default async function downloadFile(
   await downloadToFs(completeFilename, fileStreamPromise, support, isFirefox, abortController).catch((err) => {
     const errMessage = err instanceof Error ? err.message : (err as string);
 
-    trackFileDownloadError(userEmail, fileId, errMessage);
+    if (errMessage.includes('user aborted')) {
```
Comment (sg-gs) [inline src/app/drive/services/download.service/downloadFile.ts]: To ensure that the abort always is caught, check use:
` if (abortController.signal.aborted) {...}`

File: src/app/drive/services/download.service/downloadFile.ts [hunk 2]
```diff
@@ -125,7 +105,16 @@ export default async function downloadFile(
     support = DownloadSupport.PatchedStreamApi;
   }
 
-  trackFileDownloadStart(userEmail, fileId, itemData.name, itemData.size, itemData.type, itemData.folderId);
+  const trackingDownloadProperties: TrackingPlan.DownloadProperties = {
+    file_download_id: analyticsService.getTrackingActionId(),
+    file_id: parseInt(fileId),
```
Comment (sg-gs) [inline src/app/drive/services/download.service/downloadFile.ts]: Is this the fileId from the network (string) or the autoincremental id (unsigned int)? Unless the interface is explicitly changed, I will recommend double-checking if this is a string or an int, parseInt could fail and all the `downloadFile` function will crash

**Review by sg-gs — CHANGES_REQUESTED** [review]

File: src/app/drive/services/download.service/downloadFile.ts (hunk 1 above)
Comment (JoanMora) [reply src/app/drive/services/download.service/downloadFile.ts]: Added

File: src/app/drive/services/download.service/downloadFile.ts (hunk 2 above)
Comment (JoanMora) [reply src/app/drive/services/download.service/downloadFile.ts]: This is the network fileId. Changes added

File: src/app/drive/services/download.service/downloadFile.ts [hunk 3]
```diff
@@ -125,7 +105,16 @@ export default async function downloadFile(
     support = DownloadSupport.PatchedStreamApi;
   }
 
-  trackFileDownloadStart(userEmail, fileId, itemData.name, itemData.size, itemData.type, itemData.folderId);
+  const trackingDownloadProperties: TrackingPlan.DownloadProperties = {
+    file_download_id: analyticsService.getTrackingActionId(),
+    file_id: typeof fileId === 'string' ? parseInt(fileId) : fileId,
```
Comment (sg-gs) [inline src/app/drive/services/download.service/downloadFile.ts:110]: Parsing a string with the hex format (like the fileId from the network) is going to give you 'NaN'

**Review by sg-gs — CHANGES_REQUESTED** [review]

**Comment by sg-gs** [comment]
Environment variables are set and ready.

**Review by sg-gs — APPROVED** [review]

**Review by CandelR — APPROVED** [review]

**Pull request merged**